### PR TITLE
[PAN-1850] Publish to PyPI

### PR DESCRIPTION
**Pull request by jpantos**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
- Enables PyPI Trusted Publishing for the Common library.
- Adds version check to ensure the Github Release is the same as in pyproject.toml
- Freezes all the dependencies from poetry.lock onto the wheel package

## Related Tickets & Documents

- Closes #PAN-1850

## QA Instructions
N/A

## [optional] Are there any post deployment tasks we need to perform?
Create a Release

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -10,23 +10,84 @@ concurrency:
...
+          poetry freeze-wheel
 
       - name: Upload build artifact
         uses: actions/upload-artifact@v4
         with:
           name: common
           path: dist
 
+  publish-pypi:
+    name: Publish to Mainnet
```
Comment (danut13) [inline .github/workflows/release.yaml:65]: Wrong name I think

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -10,23 +10,84 @@ concurrency:
...
+          disable-sudo: true
+          egress-policy: audit
+
+      - name: Download build artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: common
+          path: dist
+
+      - name: Publish package distributions to PyPI
```
[the remark this answers is not in the document]
Comment (danut13) [reply .github/workflows/release.yaml]: Ah got it, now I see it. I'm not sure if that fits in the makefile tho. This is a release pipeline specific thing which does not really make sense outside of it. I would integrate it directly into the pipeline logic and remove the make target. I'm fine with both options, maybe give it a thought

[the remark this answers is not in the document]
Comment (jpantos) [reply .github/workflows/release.yaml]: I'd say it's a project specific (to be fair, I'd also prefer to have the poetry freeze logic inside the Makefile). If we were to reuse this workflow somewhere else then it should work as long as we have the same makefile command

[the remark this answers is not in the document]
Comment (juanmanuel-tirado) [reply .github/workflows/release.yaml]: We could eventually move this into the `ci-workflows`

**Review by juanmanuel-tirado — APPROVED** [review]
LG2M

**Pull request merged**